### PR TITLE
[ARITH] Subspace division

### PR DESCRIPTION
**Pull request by spectrometerHBH**
cc @tqchen

### CODE REVIEW

**Comment by tqchen** [comment]
@spectrometerHBH please rebase now that #7759 is merged. cc @hzfan @Hzfengsy @jinhongyii please also take a look when you have time.

File: src/arith/iter_affine_map.cc [hunk 1]
```diff
@@ -1028,5 +1028,358 @@ PrimExpr IterMapRewriter::VisitExpr_(const FloorModNode* op) {
...
+    }
+
+   private:
+    static IterSplitExpr GetAsSplit(const IterMapExpr& expr, const PrimExpr& extent) {
+      if (const auto* op = expr.as<IterSplitExprNode>()) {
+        return GetRef<IterSplitExpr>(op);
+      } else if (const auto* op = expr.as<IterSumExprNode>()) {
+        return IterSplitExpr(IterMark(GetRef<IterSumExpr>(op), extent));
+      } else {
+        LOG(FATAL);
```
Comment (Hzfengsy) [inline src/arith/iter_affine_map.cc:1152]: It would be great if you can add some error message here.

**Review by Hzfengsy — APPROVED** [review]
Overall LGTM except for some nits

File: src/arith/iter_affine_map.cc [hunk 2]
```diff
@@ -1086,5 +1086,302 @@ TVM_REGISTER_GLOBAL("arith.NormalizeIterMapToExpr").set_body_typed([](const Iter
...
+  DivisionResult DivideIterSplitExpr(const IterSplitExpr& expr) {
+    auto it = split_map_.find(expr);
+    if (it != split_map_.end()) {
+      // We will calculate all the splits of an IterMark's division form when we first
+      // encounter one of them. If we encounter another later, we directly return the record.
+      return it->second;
+    }
+    const Array<IterSplitExpr>& splits = collector_.mark2splits_.at(expr->source);
+    if (const auto* iter_ptr = expr->source->source.as<VarNode>()) {
+      // source is input_iter,
```
Comment (MasterJH5574) [inline src/arith/iter_affine_map.cc:1265]:
```suggestion
      // source is input_iter
```

**Review by MasterJH5574 — APPROVED** [review]

**Review by tqchen — APPROVED** [review]

File: python/tvm/arith/iter_affine_map.py [hunk 3]
```diff
@@ -128,3 +128,46 @@ def normalize_iter_map_to_expr(expr):
...
+
+    predicate : PrimExpr
+        The predicate constraints on the input iterators
+
+    require_bijective : bool
+        A boolean flag that indicates whether the bindings should be bijective
+
+    Returns
+    -------
+    results : List[List[PrimExpr]]
```
[the remark this answers is not in the document]
Comment (spectrometerHBH) [reply python/tvm/arith/iter_affine_map.py:167]: Yes. I will add that in the doc.

**Pull request merged**